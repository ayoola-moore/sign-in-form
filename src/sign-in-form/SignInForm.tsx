import axios from "axios";
import { useFormik } from "formik";
import React from "react";
import * as Yup from "yup";
import { Input, Label, RememberMe, SignIn } from "./styled";

interface IValues {
  email: string;
  password: string;
  remember: string;
}

const SignInForm = () => {
  const formik = useFormik({
    initialValues: {} as IValues,
    validationSchema: Yup.object({
      email: Yup.string().email().required("Email is required"),
      password: Yup.string()
        .min(8, "Password is too short - should be 8 chars minimum.")
        .required("Password is required"),
    }),
    onSubmit: ({ email, password, remember }: IValues) => {
      axios
        .post(`localhost:8000/api/login`, {
          email,
          password,
          remember,
        })
        .then((res) => {
          console.log(`${res} successful`);
        })
        .catch((error) => {
          console.error("API request failed");
        });
    },
  });

  return (
    <>
      <form onSubmit={formik.handleSubmit}>
        <Label htmlFor="email">Email</Label>
        <Input
          id="email"
          name="email"
          type="email"
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          value={formik.values.email}
          placeholder={"Enter your email"}
        />
        <div className={"error"}>
          {formik.touched.email && formik.errors.email ? (
            <div>{formik.errors.email}</div>
          ) : null}
        </div>
        <br />
        <Label htmlFor="password">Password</Label>
        <Input
          id="password"
          name="password"
          type="password"
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          value={formik.values.password}
          placeholder={"Enter your password"}
        />
        <div className={"error"}>
          {formik.touched.password && formik.errors.password ? (
            <div>{formik.errors.password}</div>
          ) : null}
        </div>
        <br />
        <RememberMe>
          <Input
            id="remember"
            name="remember"
            type="checkbox"
            onChange={formik.handleChange}
            onBlur={formik.handleBlur}
            value={formik.values.remember}
          />
          <label htmlFor="Remember me"> Remember me?</label>
        </RememberMe>

        <SignIn type="submit" disabled={!(formik.isValid && formik.dirty)}>
          Sign in
        </SignIn>
      </form>
    </>
  );
};

export default SignInForm;
