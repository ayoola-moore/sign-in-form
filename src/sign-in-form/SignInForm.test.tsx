import React from "react";
import renderer from "react-test-renderer";
import SignInForm from "./";

it("Sign-in form component renders correctly", () => {
  const tree = renderer.create(<SignInForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
