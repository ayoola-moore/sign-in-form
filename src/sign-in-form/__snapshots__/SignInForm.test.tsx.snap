// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign-in form component renders correctly 1`] = `
<form
  aria-label="Sign in form"
  className="sc-eCssSg fSuOCU"
  onSubmit={[Function]}
  role="form"
>
  <label
    className="sc-gsTCUz bdHVcV"
    htmlFor="email"
  >
    Email
  </label>
  <input
    className="sc-dlfnbm gtokWY"
    id="email"
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Enter your email"
    type="email"
  />
  <div
    className="error"
  />
  <br />
  <label
    className="sc-gsTCUz bdHVcV"
    htmlFor="password"
  >
    Password
  </label>
  <input
    className="sc-dlfnbm gtokWY"
    id="password"
    name="password"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Enter your password"
    type="password"
  />
  <div
    className="error"
  />
  <br />
  <div
    className="sc-hKgILt fyGWxI"
  >
    <input
      className="sc-dlfnbm gtokWY"
      id="remember"
      name="remember"
      onBlur={[Function]}
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="remember"
    >
       Remember me?
    </label>
  </div>
  <button
    className="sc-bdfBwQ gYXtGj"
    disabled={true}
    type="submit"
  >
    Sign in
  </button>
</form>
`;
